### PR TITLE
Support torch.dtype as parameter in pybind11 cpp extension.

### PR DESCRIPTION
**Pull request by Shan19900305**
Support torch.dtype as parameter in pybind11 cpp extension.
Example:
`
cpp_extension.my_ops(self, other, torch.dtype)
`

@ezyang @bdhirsh

### CODE REVIEW

**Comment by colesbury** [comment]
@ezyang - would you please review this PR?

File: torch/csrc/utils/pybind.h [hunk 1]
```diff
@@ -189,6 +191,36 @@ struct type_caster<at::Device> {
...
+    }
+    return false;
+  }
+
+  static handle cast(
+      const at::ScalarType& src,
+      return_value_policy /* policy */,
+      handle /* parent */) {
+    auto [primary_name, legacy_name] = torch::utils::getDtypeNames(src);
+    return handle(THPDtype_New(src, primary_name));
```
Comment (ezyang) [inline torch/csrc/utils/pybind.h:220]: this is the wrong function, use getTHPDtype from torch/csrc/DynamicTypes.h

**Review by ezyang — CHANGES_REQUESTED** [review]
Will accept, but need to fix

File: torch/csrc/utils/pybind.h (hunk 1 above)
Comment (Shan19900305) [reply torch/csrc/utils/pybind.h]: ok, thx.

**Comment by Shan19900305** [comment]
> Will accept, but need to fix

@ezyang  Hi Yang, pls review this PR again, thx.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**